contributors/THIRD-PARTY-EXTENSION.md

```diff
@@ -0,0 +1,119 @@
+## Introduction
+
+Welcome to the guide for developing third party extensions in the `create-eth` repository.
+
+Third party extension allows developers to **extend** base instance created by `npx create-eth@latest` and publish the extension to github that can be used by other developers via:
+
+```bash
+npx create-eth@latest -e {your-github-userName}/{extension-repo-name}:{extension-branch-name} # extension-branch-name is optional
+```
+
+It is important to understand distinction between instance and extension.
+
+- Instance is created by `npx create-eth@latest` and contains full project files.
+- Extension does not contain full project files, instead it only contains extra files, extra directories and [`*.args.mjs`](TEMPLATING.md#args-files) files that will added to the base instance created via `npx create-eth@latest` with `-e` flag.
+
+> TIP: Checkout directory structure of `eip-712` extension maintained Scaffold-ETH team at https://github.com/scaffold-eth/create-eth-extensions/tree/eip-712
+
+This document outlines the workflow for creating and testing your extensions.
+
+## Workflow Overview
+
+An developer goes through 2 phases while developing extension:
+
+- **Phase 1**: Initial Setup and Development
+  - This phase helps you create your own extension based on the changes you make to instance created by `npx create-eth@latest`
+- **Phase 2**: Local Testing and Publishing
+  - This phase helps you to try out your extension locally and see how it works when used by other developers before publishing to github
+
+### Phase 1: Initial Setup and Development
+
+1. **Clone the `create-eth` Repository:**
+
+   ```bash
+   git clone https://github.com/scaffold-eth/create-eth.git
+   cd create-eth
+   ```
+
+2. **Run the Build Script:**
+
+   ```bash
+   yarn build:dev
+   ```
+
+   This creates `cli.js` and `create-extension.js` in the `dist` directory.
+
+3. **Run the CLI to Create a New Instance:**
+
+   ```bash
+   yarn cli
+   ```
+
+   This command will create a **new base instance** which is similar to running `npx create-eth@latest`
+
+   The name mentioned for "Your project name" question will be used as the **extension name**. For example, if you provide `eip` as value to the question then the final extension name will be `eip`.
+
+4. **Develop the Extension:**
+
+   - CD into the instance directory.
+   - Make necessary changes to the instance project.
+   - Commit the changes in the instance repository.
+
+   **Some Caveats:**
+
+   - Only adding of new files, new directories is allowed while creating extension. If you try to overwrite existing files, it wont be reflected.
+   - `*.args.mjs` files: You might want to add content existing files(base instance files), for example adding new page link in Header component. `create-eth` allows injecting additional content to **certain files** with [`*.args.mjs`](TEMPLATING.md#args-files) files.
```

Comment:
Are devs also limited to *certain places* within *certain files* or can they also do stuff like adding arbitrary imports and jsx?

```suggestion
   - `*.args.mjs` files: You might want to add content to existing base instance files. For example, adding a new page link in the Header component. `create-eth` allows injecting additional content into **certain files** with [`*.args.mjs`](TEMPLATING.md#args-files) files.
```